### PR TITLE
feat(selenium): update docs and include SE_NODE_GRID_URL requirement

### LINKED ISSUE #11692: [BUG] Playwright is not running tests on Selenium Grid + Docker
**Context:**
- Playwright Version: 8.1.2
- Operating System: Linux (5.10.60.1-microsoft-standard-WSL2)
- Node.js version: 16.13.2
- Selenium Grid version: 4.1.1
- Browser: Chrome (v.97.0)
- Extra: The operating system is an image of Linux on Docker Desktop version 4.4.3 (73365) running through Windows 10.

**Bug description**

I'm trying to run some Playwright (Node.js) tests on Docker through Selenium Grid, but the tests are failing because of timeout. Chrome is apparently not starting. Locally without Docker everything is fine.

According to the [Playwright documentation](https://playwright.dev/docs/selenium-grid), it would be enough to run the tests using the following command:

> SELENIUM_REMOTE_URL=http://localhost:4444/wd/hub npx playwright test

But that is not working.

I'm building the **environment in Docker** by running the following file through Powershell:

```
$maxNodes = 1

function GetImages()
{
    docker pull selenium/hub:latest
    docker pull selenium/node-chrome:latest
}

function CreateGrid()
{
    docker network create grid
}

function CreateHub()
{
   docker run -d -p 4442-4444:4442-4444 --net grid --name hub selenium/hub:latest
}

function CreateNodes()
{
    $nodes = 1

    while($nodes -le $maxNodes)
    {
        docker run -d -P --net grid -e SE_EVENT_BUS_HOST=hub -e SE_EVENT_BUS_PUBLISH_PORT=4442 -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 -e SE_NODE_SESSION_TIMEOUT=120 selenium/node-chrome:latest
        $nodes++
    }    
}

cls
GetImages
CreateGrid
CreateHub
CreateNodes
cls
Write-Host "HUB AND NODES CREATED!!"
```

The config from playwright.config.ts is:

```
import { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
  forbidOnly: !!process.env.CI,
  retries: process.env.CI ? 1 : 0,
  timeout: 20000,
  workers:1,  
  use: {
    baseURL: 'https://www.google.com',
    viewport: { width: 1280, height: 720 },
    browserName: "chromium",
    channel: "chrome",
    headless: false    
  }
};
export default config;
```

And the test running is:

```
test('Acessar Google', async ({ page }) => {

   await page.goto('/');

   const length = await page.locator('input[type=submit]').count();
   expect(length >= 1).toBeTruthy(); 
});
```

In Docker Selenium Grid, the following is displayed:

image

The session is "empty" with nothing running. The browser has a blank page.

image

After running the tests using the above mentioned command, the result is as follows.

image

I think this is a bug. If anyone can help, I'd be grateful.

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #11692, #12793.

### CODE REVIEW

File: docs/src/selenium-grid.md [hunk 1]
```diff
@@ -3,66 +3,174 @@ id: selenium-grid
...
-# Playwright Test
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub npx playwright test
+If webdriver example does not work, look for any errors at your Selenium hub/node/standalone output and search [Selenium issues](https://github.com/SeleniumHQ/selenium/issues) for a possible solution.
+:::
+
+## Starting Selenium Grid
 
-# Playwright Library
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub node script.js
+If you run distributed Selenium Grid, Playwright needs selenium nodes to be registered with an accessible address, so that it could connect to the browsers. To make sure it works as expect, set `SE_NODE_GRID_URL` environment variable pointing to the hub when running selenium nodes.
```
Comment (yury-s) [inline docs/src/selenium-grid.md:16]: typo: works as expect -> works as expected

File: docs/src/selenium-grid.md [hunk 2]
```diff
@@ -3,66 +3,174 @@ id: selenium-grid
...
+# Start selenium node
+SE_NODE_GRID_URL="http://<selenium-hub-ip>:4444/wd/hub" java -jar selenium-server-<version>.jar node
 ```
 
-```bash python
-# With Pytest
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub pytest --browser chromium
+## Connecting Playwright to Selenium Grid
+
+To connect Playwright to **Selenium Grid 4**, set `SELENIUM_REMOTE_URL` environment variable pointing to your Selenium Grid Hub. Note this only works for Google Chrome and Microsoft Edge.
```
Comment (yury-s) [inline docs/src/selenium-grid.md:25]: nit: Note this -> Note that this

File: docs/src/selenium-grid.md [hunk 3]
```diff
@@ -3,66 +3,174 @@ id: selenium-grid
...
 
-You don't have to change your code, just use [`method: BrowserType.launch`] as usual.
+You don't have to change your code, just use your testing harness or [`method: BrowserType.launch`] as usual.
 
 When using Selenium Grid Hub, you can [skip browser downloads](./browsers.md#skip-browser-downloads).
 
-If your grid requires additional capabilities to be set (for example, you use an external service), you can use `SELENIUM_REMOTE_CAPABILITIES` environment variable to provide JSON-serialized capabilities.
+### Passing additional capabilities
+
+If your grid requires additional capabilities to be set (for example, you use an external service), you can set `SELENIUM_REMOTE_CAPABILITIES` environment variable to provide JSON-serialized capabilities.
```
Comment (yury-s) [inline docs/src/selenium-grid.md:49]: nit: maybe link SELENIUM_REMOTE_CAPABILITIES to a selenium doc?

**Review by yury-s — APPROVED** [review]

File: docs/src/selenium-grid.md (hunk 1 above)
Comment (Mone71) [reply docs/src/selenium-grid.md:16]: Thanks

File: docs/src/selenium-grid.md (hunk 3 above)
Comment (Mone71) [reply docs/src/selenium-grid.md:49]: Thanks

File: docs/src/selenium-grid.md (hunk 2 above)
Comment (Mone71) [reply docs/src/selenium-grid.md:25]: Thanks

File: docs/src/selenium-grid.md (hunk 3 above)
Comment (dgozman) [reply docs/src/selenium-grid.md:49]: It is a playwright thing, does not exist in selenium webdriver.

**Pull request merged**